time_execution/backends/elasticsearch.py

```diff
@@ -22,15 +29,15 @@ def __init__(
         # setup the client
         self.client = Elasticsearch(hosts=hosts, *args, **kwargs)
 
-        # ensure the index is created
-        try:
-            self._setup_index()
-        except TransportError as exc:
-            logger.error("index setup error %r", exc)
-        try:
-            self._setup_mapping()
-        except TransportError as exc:
-            logger.error("mapping setup error %r", exc)
+        if not skip_index_creation:
```

Comment:
minor: may be a matter of taste, but I think that negated conditions(and especially double negations) are slightly more difficult to read. so, I would rather use `if create_index:` instead of `if not do_not_create_index:`